src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2262,10 +2541,99 @@ bool TheoryStringsRewriter::checkEntailArith(Node a, bool strict)
       return true;
     }
     // TODO (#1180) : abstract interpretation goes here
+
+    // over approximation O/U
+
+    // O( x <op> y ) -> O( x ) <op> O( y )
+    // O( len( x ) ) -> ...
+    // O( len( int.to.str( x ) ) ) -> ...
+    // O( len( str.substr( x, n1, n2 ) ) ) -> O( n2 ) | O( len( x ) )
+    // O( len( str.replace( x, y, z ) ) ) -> 
+    //   O( len( x ) ) + O( len( z ) ) - U( len( y ) )
+    // O( indexof( x, y, n ) ) -> O( len( x ) ) - U( len( y ) )
+    // O( str.to.int( x ) ) -> ...
+
+    // U( x <op> y ) -> U( x ) <op> U( y )
+    // U( len( x ) ) -> ...
+    // U( len( int.to.str( x ) ) ) -> 1
+    // U( len( str.substr( x, n1, n2 ) ) ) -> 
+    //   min( U( len( x ) ) - O( n1 ), U( n2 ) )
+    // U( len( str.replace( x, y, z ) ) ) -> 
+    //   U( len( x ) ) + U( len( z ) ) - O( len( y ) ) | 0
```

Comment:
One rule that could be helpful here: `U( len( str.concat( x1, ..., xn ) ) ) =  U( len( x1 ) ) + ... U( len( xn ) )` (related to [this comment](https://github.com/CVC4/CVC4/pull/1337/files/560b4f72993533277f04a3e68e4c4630976f2e44#r150697475)).

Reply:
Ah, so I am assuming that U( t ) is such that t is in rewritten form, where notice that : 
str.len( str.++( x1, ..., xn ) ) ----> str.len( x1 ) + ... + str.len( xn )
Hence we don't have to care about this case.